### PR TITLE
Refs [TASK][35857] Create ui screen favorite

### PR DESCRIPTION
**Pull request by chiennn19122000**
## Related Tickets
- [#Task 35857:  Create ui screen favorite](https://edu-redmine.sun-asterisk.vn/issues/35857)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteFragment.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    override fun showProducts(products: List<Product>) {
+        favoriteProductAdapter.replaceData(products)
+    }
+
+    private fun itemFavoriteClicked(product: Product) {
+        fragmentManager?.addFragment(
+            R.id.frameContainer,
+            ProductDetailFragment.getInstance(product)
+        )
+
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteFragment.kt]: .

File: app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoritePresenter.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.sun.makeupwindow.data.source.utils.OnDataLoadCallback
+import java.lang.Exception
+
+class FavoritePresenter (
+    private val view: FavoriteContract.View,
+    private val favoriteProductRepository: FavoriteProductRepository
+) : FavoriteContract.Presenter {
+
+    override fun getProducts() {
+        favoriteProductRepository.getProducts(object : OnDataLoadCallback<List<Product>> {
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoritePresenter.kt]: thiếu showLoading trước khi hide nhé

File: app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteProductViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import com.sun.makeupwindow.base.BaseViewHolder
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.utlis.loadImage
+import kotlinx.android.synthetic.main.item_favorite.view.*
+
+class FavoriteProductViewHolder(
+    private val itemView: View,
+    onItemClick: (Product) -> Unit
+) : BaseViewHolder<Product>(itemView) {
+    private var itemData: Product? = null
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteProductViewHolder.kt]: .

File: app/src/main/java/com/sun/makeupwindow/ui/main/MainActivity.kt [hunk 4]
```diff
@@ -25,6 +26,7 @@ class MainActivity : BaseActivity() {
         BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
             when (menuItem.itemId) {
                 R.id.menuHome -> openFragment(HomeFragment.getInstance())
+                R.id.menuFavorite ->openFragment(FavoriteFragment.getInstance())
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/makeupwindow/ui/main/MainActivity.kt]: format code

File: app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteFragment.kt (hunk 1 above)
Comment (chiennn19122000) [reply app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteFragment.kt]: chỗ nãy bị s à anh

File: app/src/main/res/layout/fragment_favorite.xml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="@+id/viewHeader" />
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/recyclerFavoriteProduct"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/dp_8"
+            android:layout_marginStart="@dimen/dp_10"
+            android:orientation="vertical"
```
Comment (phucsaker99) [inline app/src/main/res/layout/fragment_favorite.xml]: mặc định vertical ko nhất thiết phải khai báo nhé

File: app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteFragment.kt (hunk 1 above)
Comment (phucsaker99) [reply app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteFragment.kt]: e thừa 1 line à

File: app/src/main/res/layout/fragment_favorite.xml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            android:text="@string/title_favorite"
+            android:textColor="@android:color/white"
+            android:textSize="@dimen/sp_25"
+            android:textStyle="bold"
+            app:layout_constraintBottom_toBottomOf="@+id/viewHeader"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="@+id/viewHeader" />
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/recyclerFavoriteProduct"
```
Comment (phucsaker99) [inline app/src/main/res/layout/fragment_favorite.xml]:
```suggestion
            android:id="@+id/recyclerFavoriteProducts"
```

File: app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteProductViewHolder.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    override fun onBind(item: Product) {
+        itemData = item
+        itemView.run {
+            item.imageLink?.let {
+                imageFavoriteProduct.loadImage(it)
+            }
+            textName.text = item.name
+            textPrice.text = item.price + " " + item.priceSign
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/makeupwindow/ui/favorite/FavoriteProductViewHolder.kt]: ai cho + string kiểu này :v

File: app/src/main/res/layout/fragment_favorite.xml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        <TextView
+            android:id="@+id/textHome"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/dp_20"
+            android:text="@string/title_favorite"
+            android:textColor="@android:color/white"
+            android:textSize="@dimen/sp_25"
+            android:textStyle="bold"
+            app:layout_constraintBottom_toBottomOf="@+id/viewHeader"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_favorite.xml]:
```suggestion
            app:layout_constraintBottom_toBottomOf="@id/viewHeader"
```

File: app/src/main/res/layout/fragment_favorite.xml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/recyclerFavoriteProducts"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/dp_8"
+            android:layout_marginStart="@dimen/dp_10"
+            app:layoutManager="androidx.recyclerview.widget.GridLayoutManager"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="@+id/viewHeader"
+            app:layout_constraintTop_toBottomOf="@+id/viewHeader"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_favorite.xml]:
```suggestion
            app:layout_constraintStart_toStartOf="@id/viewHeader"
            app:layout_constraintTop_toBottomOf="@id/viewHeader"
```
trc có dặn là tìm hiểu r mà vẫn mắc lại nhỉ

**Pull request merged**